core/integration/secret_test.go

```diff
@@ -133,8 +133,24 @@ func TestSecretPlaintext(t *testing.T) {
 	require.NoError(t, err)
 	defer c.Close()
 
+	//nolint:staticcheck // SA1019 We want to test this API while we support it.
 	plaintext, err := c.Directory().
 		WithNewFile("TOP_SECRET", "hi").File("TOP_SECRET").Secret().Plaintext(ctx)
 	require.NoError(t, err)
 	require.Equal(t, "hi", plaintext)
 }
+
+func TestNewSecret(t *testing.T) {
+	ctx := context.Background()
+	c, err := dagger.Connect(ctx)
+	require.NoError(t, err)
+	defer c.Close()
+
+	secretValue := "very-secret-text"
+
+	s := c.SetSecret("aws_key", secretValue)
+
+	exitCode, err := c.Container().From("alpine:3.16.2").WithSecretVariable("AWS_KEY", s).WithExec([]string{"sh", "-c", "test \"$AWS_KEY\" = \"very-secret-text\""}).ExitCode(ctx)
```

Comment:
nit: better to wrap long lines for readabilty:

```suggestion
	exitCode, err := c.Container().From("alpine:3.16.2").
		WithSecretVariable("AWS_KEY", s).
		WithExec([]string{"sh", "-c", "test \"$AWS_KEY\" = \"very-secret-text\""}).
		ExitCode(ctx)
```